### DISCUSSION TITLE
PDK should be more modular

### DISCUSSION THREAD

**Discussion by trevor-vaughan** [puppetlabs/pdk#1011] [Feature Request]
## Use Case
As a user, I would like to be able to easily use some of the features of PDK without having to drink the Kool-Aid(tm)

Specifically, I would like to be able to use the module building and code validation capabilities from within a system-native development stack (at my peril for ruby and puppet compatibility).

## Describe the Solution You Would Like
I believe that parts of the stack that aren't directly related to user-oriented PDK tasks should be split into separate Gems to ensure that those parts can be independently upgraded with a minimum of external requirements.

## Describe Alternatives You've Considered
Currently, I pull in the `pdk` gem and perform brain surgery to use the internal classes. This is both non-optimal and fragile for things that should be able to be run via CI systems without needing to install the entire PDK stack.

## Additional Context
I understand that I'm a relatively rare power-user use case but I think that supporting both the PDK-pure and those that have complex legacy stacks is a win for everyone. Also, modularity is almost always a good thing in terms of maintenance and testing.

**Comment by logicminds** [comment]
👏 I'll be making an RFC soon for this cause I like power too.
